### PR TITLE
Allow overriding memory limit on docker build/run

### LINKED ISSUE #6342: Cannot override default Docker memory limit
## Issue Description

For some workloads, the default memory limit for Docker `build` and `run` tasks of 1GB is not enough. It would be nice to override this in those tasks (perhaps "hidden away" under the **Advanced** section). Error handling could be delegated to Docker itself that will fail if not enough memory is available below the set limit.

### PR DESCRIPTION
**Pull request by heaths**
Fixes #6342

### CODE REVIEW

File: Tasks/Docker/task.json [hunk 1]
```diff
@@ -294,6 +294,14 @@
...
             "helpMarkDown": "Working directory for the Docker command.",
             "groupName": "advanced"
+        },
+        {
+            "name": "memory",
+            "type": "string",
+            "label": "Memory limit",
+            "visibleRule": "action = Build an image || action = Run an image",
+            "helpMarkDown": "The maximum amount of memory available to the container as a integer with optional suffixes like '2GB'.",
+            "groupName": "advanced"
```
Comment (jikuma) [inline Tasks/Docker/task.json]: Is your build passing? I think inputs and dependents inputs should be in the same group?.

Comment (heaths) [reply Tasks/Docker/task.json]: Not sure what dependent inputs you're talking about.

Comment (jikuma) [reply Tasks/Docker/task.json]: Visibility of 'memory' is dependent on 'action'. While action is in no group 'memory' is in advance group. VSTS task compile fails if both the inputs are in the different groups. I am not sure what will happen if one of the input is in no specified group. If there is no compilation error, its good. I just wanted you to check this.

Comment (heaths) [reply Tasks/Docker/task.json]: I, of course, compiled and ran both the existing and new tests I wrote. This is written like the other task options dependency on `action`.

Comment (heaths) [reply Tasks/Docker/task.json]: ...that said, have you considered build validation via TravisCI, AppVeyor, and/or other? I see you have a _.vsts-cj.yml_ file as well. Is that not configured for PRs? I was surprised to see that there was no PR validation when the environment is fairly self-contained (except for requiring npm 3.x when I had 5.x that does not work for this repo). That would help provide some assurance no one breaks anything and you don't have to rely on manual testing or contributors' word they built and tested...even though I did. :)

**Pull request merged**